docs/exceptions.md

```diff
@@ -164,13 +195,29 @@ to return `Maybe v`. In 100% pure code, use one of these ways to handle errors:
 * `MaybeT`, `ExceptT e`
 * `CatchT`
 
+*Comment:* neither of these examples are of errors or exceptions! They are
+examples of normal values in the images of pure functions. The `Map` does not
+make any guarantee to that any particular key is in the map! Nor does the parser
+guarantee that every input will be recognized! These facts are reflected in the
+types. This has *nothing at all* to do with errors or exceptions. We need to
```

Comment:
I'd say that "parse errors" do have something to do with "errors", they are called "parse errors" for a reason. And they might as well have something to do with "exceptions", because it'd be valid to represent them as `ExceptT`, which has got `Except` in its name. (IO exceptions would work too, but parsers are normally pure).

Reply:
@avieth I think that you're doing more work than it is truly optimal. We need to agree on the definitions first :)

"error" means "something that is not default", so "parse errors" are "errors" just because it's too long to call them "completely normal execution paths that didn't return the parse result though". No finding the key in hashmap is completely normal too, though it's called a "lookup fail" or "lookup error" because "not finding the key in hashmap" is too long.

Reply:
> "parse errors" are "errors" just because it's too long to call them "completely normal execution paths that didn't return the parse result though"

You don't have to call them that. How about "the image of `parse x`". A parse failure is just as much in the image of `parse` as a parse success. 

> I think that you're doing more work than it is truly optimal. We need to agree on the definitions first :)

Sure, I'm showing where/how/why I disagree with the definitions.